### PR TITLE
Implement RWops write functionality

### PR DESCRIPTION
**Pull request by AI0867**

### CODE REVIEW

File: src/filesystem_sdl.cpp [hunk 1]
```diff
@@ -114,11 +171,31 @@ static size_t SDLCALL ifs_read(struct SDL_RWops *context, void *ptr, size_t size
...
+
+	const std::streampos before = ofs->tellp();
+	ofs->write(static_cast<const char*>(ptr), num * size);
+	const std::streampos after = ofs->tellp();
+
+	const int written = (after - before) / size;
+
+	// Fail sticks unless we clear it. Bad is an actual I/O error
+	if(!ofs->bad())
+		ofs->clear();
```
Comment (jyrkive) [inline src/filesystem_sdl.cpp]: Please remove this part. `std::basic_ostream::write()` isn't supposed to set failbit, ever.

> Also, unlike the formatted output functions, this function does not set the failbit on failure.

http://en.cppreference.com/w/cpp/io/basic_ostream/write

File: src/filesystem_sdl.cpp [hunk 2]
```diff
@@ -114,11 +171,31 @@ static size_t SDLCALL ifs_read(struct SDL_RWops *context, void *ptr, size_t size
...
+
+	return static_cast<int>(written);
```
Comment (jyrkive) [inline src/filesystem_sdl.cpp]: Please remove the unnecessary cast.

File: src/filesystem_sdl.cpp [hunk 3]
```diff
@@ -63,30 +95,38 @@ static Sint64 ifs_size (struct SDL_RWops * context) {
...
-	std::ios_base::seekdir seekdir;
+	std::streampos len = ofs->tellp();
+
+	ofs->seekp(orig);
+
+	return len;
+}
+
+typedef std::pair<Sint64, std::ios_base::seekdir> offset_dir;
+static offset_dir translate_seekdir(Sint64 offset, int whence){
```
Comment (jyrkive) [inline src/filesystem_sdl.cpp]: You're missing a space before the opening brace.

File: src/SDL_SavePNG/savepng.h [hunk 4]
```diff
@@ -7,16 +7,7 @@
...
- * Save an SDL_Surface as a PNG file.
- *
- * Returns 0 success or -1 on failure, the error message is then retrievable
- * via SDL_GetError().
- */
-#define SDL_SavePNG(surface, file) \
-	SDL_SavePNG_RW(surface, SDL_RWFromFile(file, "wb"), 1)
-//
-//TODO: filesystem::load_RWops is only for reading, would like a writing version also
+
```
Comment (jyrkive) [inline src/SDL_SavePNG/savepng.h]: Do we really need this last commit?

This isn't our code, it's from https://github.com/driedfruit/SDL_SavePNG/blob/defbe6f77dace9cf4640e900f887cf8b022d55e3/savepng.h

I'd like to keep it as close to upstream as possible.

Comment (AI0867) [reply src/SDL_SavePNG/savepng.h]: This bit has been edited repeatedly in relation to our own RWops: https://github.com/wesnoth/wesnoth/commits/master/src/SDL_SavePNG/savepng.h

**Pull request merged**